homeassistant/components/alarmdecoder/sensor.py

```diff
@@ -33,6 +43,15 @@ async def async_added_to_hass(self) -> None:
             )
         )
 
+    @property
+    def device_info(self):
+        return {
+            "identifiers": {(DOMAIN, self._client.serial_number)},
+            "manufacturer": "NuTech",
+            "serial_number": self._client.serial_number,
+            "sw_version": self._client.version_number,
+        }
+
```

Comment:
That is a lot of duplicate code.
I suggest that you create a base entity in a new `entity.py` module.

```python
# in entity.py
class AlarmDecoderEntity(Entity):
    @property
    def device_info(self):
...
# in sensor.py
class AlarmDecoderSensor(AlarmDecoderEntity, SensorEntity):

```